Begin final review of text (#382)
worldbank/d4di: chapters/introduction.tex

An empirical revolution has changed the face of research economics rapidly over the last decade.
%had to remove cite {\cite{angrist2017economic}} because of full page width
Today, especially in the development subfield, working with raw data --
whether collected through surveys or acquired from ``big'' data sources like sensors, satellites, or call data records --
is a key skill for researchers and their staff.
At the same time, the scope and scale of empirical research projects is expanding:
more people are working on the same data over longer timeframes.
As the ambition of development researchers grows, so too has the complexity of the data
on which they rely to make policy-relevant research conclusions.
Yet there are few guides to the conventions, standards, and best practices
that are fast becoming a necessity for empirical research.
This book aims to fill that gap.

This book is targeted to everyone who interacts with development data:
graduate students, research assistants, policymakers, and empirical researchers.
It covers data workflows at all stages of the research process, from design to data acquisition and analysis.

> This book is not sector-specific; it will not teach you econometrics, or how to design an impact evaluation.

Reviewer: ```suggestion
Its content is not sector-specific; it will not teach you econometrics, or how to design an impact evaluation.
```

interchangeably to refer to Stata code throughout.
Stata is primarily a scripting language for statistics and data,
meaning that its users often come from economics and statistics backgrounds
and understand Stata to be encoding a set of tasks as a record for the future.
We believe that this must change somewhat:
in particular, we think that practitioners of Stata
must begin to think about their code and programming workflows
just as methodologically as they think about their research workflows,
and that people who adopt this approach will be dramatically
more capable in their analytical ability.
This means that they will be more productive when managing teams,
and more able to focus on the challenges of experimental design
and econometric analysis, rather than spending excessive time
re-solving problems on the computer.
To support this goal, this book also includes
an introductory Stata Style Guide
that we use in our work, which provides
some new standards for coding so that code styles
can be harmonized across teams for easier understanding and reuse of code.
Stata has relatively few resources of this type available,
and the ones that we have created and shared here

> we hope will be an asset to all its users.

Reviewer: Was this caused by merging different branches? I'm pretty sure this section was heavily edited in the last version of the intro I read.

Author: 2e25f2ca1bb6b522023a56f58f99d55a57c2b181 , among the other comments

handling data processing and analytical tasks.

\section{Adopting reproducible tools}

> We will provide free, open-source, and platform-agnostic tools wherever possible,

Reviewer: I think this discussion should come here:

We assume througout all of this book that you are going to do nearly all of your data work though code.
It may be possible to perform all relevant tasks through the user interface in some statistical software, 
or even through less field-specific software such as Excel.
However, we strongly advise against it.
The reason for that are the transparency, reproducibility and credibility principles discussed in chapter 1.
Writing code creates a record of every task you performed. 
It also prevents direct interaction with the data files that could lead to non-reproducible steps.
Think of the code as a recipe to create your results:
other people can follow it,
reproduce it,
and even disagree with your the amount of spices you added
(or some of your coding decisions).
For these reasons, code is now considered an essential component of a research output.

Reviewer: And by "here" I mean the first paragraph of this subsection.

Reviewer: This is the first step to solve #355

Author: 50ca90d3cd1c2839d4bc5a0d9f26a110459bc812

and point to more detailed instructions when relevant.
Stata is the notable exception here due to its current popularity in development economics.\sidenote{
\url{https://aeadataeditor.github.io/presentation-20191211/\#9}}
Most tools have a learning and adaptation process,
meaning you will become most comfortable with each tool
only by using it in real-world work.
Get to know them well early on,
so that you do not spend a lot of time learning through trial and error.

Stata is the notable exception here due to its current popularity in development economics.
We focus on Stata-specific tools and instructions in this book.
Hence, we will use the terms ``script'' and ``do-file''

> interchangeably to refer to Stata code throughout.

Reviewer: We will not!

> Stata is primarily a scripting language for statistics and data,

Reviewer: Deja-vu of a conversation about Stata being the software and not the language?

it should not require arcane reverse-engineering
to figure out what a code chunk is trying to do.
\textbf{Style}, finally, is the way that the non-functional elements of your code convey its purpose.
Elements like spacing, indentation, and naming (or lack thereof) can make your code much more
(or much less) accessible to someone who is reading it for the first time and needs to understand it quickly and correctly.

> 

Reviewer: Add here (also addressing #355):

As you gain experience in coding and get more confident with the way you implement these suggestions, you will feel more empowered to apply critical thinking to the way you handle data. For example, you will be able to predict which section of your script are more likely to create errors. This may happen intuitively, but you will improve much faster as a coder if you do it purposefully. Ask yourself, as you write code and explore results: Do I believe this number? What can go wrong in my code? How will missing values be treated in this command? What would happen if more observations would be added to the data set? Can my code be made more efficient or easier to understand?

Author: 7debf2059bb9053a0beb6aca3038451cf23a4aee

> /subsection{Code examples}

Reviewer: ```suggestion
\subsection{Code examples}
```
